### ISSUE TITLE
operator precedence of return and or

### ISSUE THREAD

**Issue by timbasel** [vlang/v#4454]
**V version:** V 0.1.26 dee6434
**OS:** Arch Linux

**What did you do?**
I want to return the value of an option.
```v
pub fn (flags []Flag) get_bool_or(name string, value bool) bool {
	return flags.get_bool(name) or { return value }
}
```

**What did you expect to see?**
Function should return the value of `flags.get_bool(name)` in case of success. If an error is returned the provided default value should be returned.
 
**What did you see instead?**
```
/home/timbasel/.cache/v/v.tmp.c: In function ‘array_cli__Flag_get_bool_or’:
/home/timbasel/.cache/v/v.tmp.c:7429:9: error: expected expression before ‘Option_bool’
 7429 |  return Option_bool tmp1 = array_cli__Flag_get_bool(flags, name);
      |         ^~~~~~~~~~~
...
```
**Currently possible workaround**
```v
pub fn (flags []Flag) get_bool_or(name string, or_value bool) bool {
	value := flags.get_bool(name) or { return or_value }
	return value
}
```

**Comment by ayazhafiz** [comment]
This happens because `return` is always written before the return expression is evaluated:

https://github.com/vlang/v/blob/b5fda6e410d2618ea801e8d1ee1be56308a94eee/vlib/v/gen/cgen.v#L1904-L1905

I think this would require a fairly large refactor of return statement codegen, either by passing a return context to nested expressions or not writing the return value until the nested expressions are generated.

**Comment by spaceface777** [comment]
Closed in favor of #6395 which contains the same issue, and is more recent.
